### PR TITLE
Fish 9730 fixing authorization tck tests for custom policy

### PR DESCRIPTION
**Pull request by breakponchito**
This is to fix failing test from the authorization TCK that requires a pre processing of custom factories for the PolicyConfigurationFactory provider and PolicyFactory provider
### Testing Performed

Execution of authorization TCK with the following results:
`

[INFO] Reactor Summary for Jakarta Authorization TCK - main 3.0.0:
[INFO]
[INFO] Jakarta Authorization TCK - main ................... SUCCESS [ 19.194 s]
[INFO] common ............................................. SUCCESS [  4.471 s]
[INFO] Authorization TCK Signature Test ................... SUCCESS [  6.343 s]
[INFO] app-custom-policy .................................. SUCCESS [ 16.106 s]
[INFO] app-custom-policy2 ................................. FAILURE [ 15.753 s]
[INFO] app-custom-policyfactory ........................... SUCCESS [ 15.726 s]
[INFO] app-custom-trace-policy ............................ SUCCESS [ 16.283 s]
[INFO] app-custom-trace-policyconfiguration ............... SUCCESS [ 16.556 s]
[INFO] app-custom-trace-policyconfigurationfactory ........ SUCCESS [ 14.795 s]
[INFO] app-policy-use-within-servlet ...................... SUCCESS [ 15.567 s]
[INFO] app-permissions-equals-hashcode .................... SUCCESS [ 14.254 s]
[INFO] app-servlet-constraints ............................ SUCCESS [ 17.516 s]
[INFO] app-ejb-constraints ................................ FAILURE [ 15.869 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:08 min
[INFO] Finished at: 2024-09-26T17:18:00-06:00

`
### Testing Environment

azul jdk 21, maven 3.9.9, ubuntu 20.04

### CODE REVIEW

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/WebBundleDescriptor.java [hunk 1]
```diff
@@ -79,6 +79,8 @@ public abstract class WebBundleDescriptor extends CommonResourceBundleDescriptor
     protected boolean conflictEntityManagerReference = false;
     protected boolean conflictEntityManagerFactoryReference = false;
 
+    private ClassLoader applicationClassLoader;
```
Comment (lprimak) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/WebBundleDescriptor.java:82]: Might need a `WeakReference<ClassLoader>` here.

**Comment by lprimak** [comment]
This needs to be checked against class loader leaks.
use `appserver/tests/payara-samples/classloader-data-api/` to check

**Comment by breakponchito** [comment]
I tested ands it seems the classloader is retained, still testing by doing deploy and undeploy of my authorization reproducer
image
now checking with weakreference

**Review by luiseufrasio — APPROVED** [review]
LGTM!

**Pull request merged**